test/e2e/dataset-delete.spec.js

```diff
@@ -490,13 +494,171 @@ describe('Dataset delete tests', () => {
+
+
+            const deleteResponse = await requester
+                .delete(`/api/v1/dataset/by-user/${USERS.USER.id}`)
+                .send();
+
+            deleteResponse.status.should.equal(401);
+            deleteResponse.body.should.have.property('errors').and.be.an('array');
+            deleteResponse.body.errors[0].should.have.property('detail').and.equal(`Unauthorized`);
+        });
+
+        it('Deleting all dataset of an user as not that same user or an admin should return a 403 error', async () => {
+            mockGetUserFromToken(USERS.MANAGER);
+            
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+
+
+            const deleteResponse = await requester
+                .delete(`/api/v1/dataset/by-user/${USERS.USER.id}`)
+                .set('Authorization', `Bearer abcd`)
+                .send();
+
+            deleteResponse.status.should.equal(403);
+            deleteResponse.body.should.have.property('errors').and.be.an('array');
+            deleteResponse.body.errors[0].should.have.property('detail').and.equal(`Forbidden`);
+        });
+
+        it('Deleting a dataset owned by the user as an ADMIN should return all deleted datasets', async () => {
+            mockGetUserFromToken(USERS.ADMIN);
+
+            nock(process.env.GATEWAY_URL)
+            .get(`/auth/user/${USERS.USER.id}`)
+            .reply(200, {
+                data: {
+                    ...USERS.USER,
+                }
+            });
+            
+            const fakeDatasetOne = await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            const fakeDatasetTwo = await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            const fakeDatasetThree = await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            prepareNockForDelete(fakeDatasetOne)
+            prepareNockForDelete(fakeDatasetTwo)
+            prepareNockForDelete(fakeDatasetThree)
+
+
+            const deleteResponse = await requester
+                .delete(`/api/v1/dataset/by-user/${USERS.USER.id}`)
+                .set('Authorization', `Bearer abcd`)
+                .send();
+
+            deleteResponse.status.should.equal(200);
+            deleteResponse.body.should.have.property('data').with.lengthOf(3);
+
+            const datasets = deserializeDataset(deleteResponse);
+            const datasetIds = datasets.map((dataset) => dataset.id);
+            datasetIds.should.contain(fakeDatasetOne._id);
+            datasetIds.should.contain(fakeDatasetTwo._id);
+            datasetIds.should.contain(fakeDatasetThree._id);
+
+            let getResponse = await requester
```

Comment:
this is ok, but I'd rather see you query the db directly. You are validating that they are gone from the db, not that an endpoint returns them.